tensorflow/contrib/periodic_resample/python/ops/periodic_resample_op.py

```diff
@@ -21,10 +21,15 @@
 
 from tensorflow.contrib.periodic_resample.python.ops import gen_periodic_resample_op
 
-from tensorflow.contrib.periodic_resample.python.ops.gen_periodic_resample_op import periodic_resample
+from tensorflow.contrib.periodic_resample.python.ops.gen_periodic_resample_op import periodic_resample, periodic_resample_op_grad
 
 from tensorflow.contrib.util import loader
+from tensorflow.python.framework import ops
 from tensorflow.python.platform import resource_loader
 
 _periodic_resample_op = loader.load_op_library(
     resource_loader.get_path_to_datafile('_periodic_resample_op.so'))
+
+@ops.RegisterGradient("PeriodicResample")
+def _periodic_resample_grad_cc(op, grad):
+  return periodic_resample_op_grad(grad, op.inputs[0].shape, op.get_attr('shape'))
```

Comment:
I think you use the tf.shape op here for the second argument, in case the shape is not known at graph construction time.

Reply:
Don't sure this is possible - periodic_resample uses special "adjustable dimension", indices for which computed differently from indices of other dimensions. Adjustable dimenstion is marked by -1 value in shape attribute, and this information is lost when considering gradient tensor shape...